src/explore-education-statistics-common/src/modules/table-tool/components/FilterHierarchyOptions.tsx

```diff
@@ -1,78 +1,155 @@
+        styles.option,
+        {
+          [styles.option__expanded]: isExpanded,
+        },
+      )}
     >
-      <div className="govuk-!-margin-left-8 govuk-!-margin-bottom-2">
-        {hideLowerTotals &&
-        level !== 0 &&
-        optionLabel.toLocaleLowerCase() === 'total' &&
-        hideLowerTotals ? null : (
-          <FormField
-            label={optionLabel}
-            id={`${name}-${optionId}`}
-            // @ts-expect-error  `value` required to make checkboxes unique
-            value={optionId}
-            name={name}
-            as={FormCheckbox}
-            checked={selectedValues.includes(optionId)}
-            disabled={disabled}
-          />
-        )}
-        {childOptionIds.map(childOptionId => (
-          <FilterHierarchyOptions
-            optionId={childOptionId}
-            key={childOptionId}
-            optionLabelsMap={optionLabelsMap}
-            filterHierarchy={filterHierarchy}
-            name={name}
-            disabled={disabled}
-            tiersTotal={tiersTotal}
-            selectedValues={selectedValues}
-            level={level + 1}
-          />
-        ))}
+      <div className={styles.checkbox}>
+        <FormField
+          hint={optionTree.filterLabel}
+          label={optionTree.label}
+          id={`${name}-${optionTree.value}`}
+          // @ts-expect-error  `value` required to make checkboxes unique
+          value={optionTree.value}
+          name={name}
+          as={FormCheckbox}
+          checked={selectedValues.includes(optionTree.value)}
+          disabled={disabled}
+        />
       </div>
+      {!!optionTree.options?.length && (
+        <DetailsMenu
+          summary={`${
+            isExpanded ? 'Close' : 'Show'
+          } ${optionTree.childFilterLabel?.toLocaleLowerCase()}`}
+          open={isExpanded}
+          onToggle={() => toggleOptions(optionTree.value)}
+          className={styles.detailsMenu}
+        >
+          <div>
+            {optionTree.options.length > 6 && (
+              <div className={styles.search}>
+                <FormTextSearchInput
+                  id={`${name}-search`}
+                  name={`${name}-search`}
+                  label={`Search ${
+                    optionTree.label
+                  } (${optionTree.childFilterLabel?.toLocaleLowerCase()})`}
+                  width={20}
+                  onChange={event => setSearchTerm(event.target.value)}
+                  onKeyPress={event => {
+                    if (event.key === 'Enter') {
+                      event.preventDefault();
+                    }
+                  }}
+                />
+              </div>
+            )}
+            <div>
+              {filteredOptions.length > 1 && (
+                <ButtonText
+                  className={styles.selectAllButton}
+                  onClick={toggleSelectAll}
+                >
+                  {hasAllSelected ? 'Unselect' : 'Select'} all{' '}
```

Comment:
These buttons also need some screen reader text to differentiate them.

Reply:
I've added some visually hidden text which should read as follows
```tsx
"Unselect all 2 options" // (screen)
"Unselect all 2 [sector subject area options for entry level]" // (screen reader)
```